src/common/transformations/include/transformations/common_optimizations/matmul_split_decomposition.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+class TRANSFORMATIONS_API MatmulGatherDecomposition;
+
+}  // namespace pass
+}  // namespace ov
+
+/**
+ * @ingroup ov_transformation_common_api
+ * @brief MatmulGatherDecomposition transformation matches following graph:
+ *
+ *         +----------+
+ *         |  input   |
+ *         +----------+
+ *              |
+ *              v
+ *         +----------+
+ *         |  MatMul  |
+ *         +----------+
+ *              |
+ *              v
+ *         +------------+
+ *         | Some nodes |
+ *         +------------+
+ *              |
+ *              v
+ *         +-----------------------+
+ *         |       Transpose       |
+ *         +-----------------------+
+ *          |          |          |
+ *          v          v          v
+ *     +-------+   +-------+   +-------+
+ *     |Gather |   |Gather |   |Gather |
+ *     +-------+   +-------+   +-------+
+ * and replaces with:
+ *
+ *         +-----------------------+
+ *         |       input           |
+ *         +-----------------------+
+ *          |          |          |
+ *          v          v          v
+ *     +-------+   +-------+   +-------+
+ *     |MatMul |   |MatMul |   |MatMul |
+ *     +-------+   +-------+   +-------+
+ *          |          |          |
+ *          v          v          v
+ *     +-------+   +-------+   +-------+
+ *     |Nodes  |   |Nodes  |   |Nodes  |
```

Comment:
Actually we have something similar
it's called TransposeSinking transformation, this is actually a set of transformations which allow to move Transpose node through other nodes
https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/transpose_sinking/ts_general.cpp
the list of the supported ops:
<img width="176" alt="{95F9113C-17D1-494D-BF8E-BEBEE578525A}" src="https://github.com/user-attachments/assets/9d9cd09e-ea90-4a41-b8d5-1caba78e5710">

I believe we need to reuse it.
Please feel free to contact me if additional details  needed.